packages/material-ui/src/SwipeableDrawer/SwipeableDrawer.js

```diff
@@ -36,6 +36,83 @@ function getTranslate(currentTranslate, startLocation, open, maxTranslate) {
   );
 }
 
+function getDomTreeShapes(element, rootNode) {
+  // adapted from https://github.com/oliviertassinari/react-swipeable-views/blob/7666de1dba253b896911adf2790ce51467670856/packages/react-swipeable-views/src/SwipeableViews.js#L129
+  let domTreeShapes = [];
+
+  while (element && element !== rootNode) {
+    const style = window.getComputedStyle(element);
+
+    if (
+      // Ignore the scroll children if the element is absolute positioned.
+      style.getPropertyValue('position') === 'absolute' ||
+      // Ignore the scroll children if the element has an overflowX hidden
+      style.getPropertyValue('overflow-x') === 'hidden'
+    ) {
+      domTreeShapes = [];
+    } else if (
+      (element.clientWidth > 0 && element.scrollWidth > element.clientWidth) ||
+      (element.clientHeight > 0 && element.scrollHeight > element.clientHeight)
+    ) {
+      // Ignore the nodes that have no width.
+      // Keep elements with a scroll
+      domTreeShapes.push({
+        element,
+        scrollWidth: element.scrollWidth,
+        scrollHeight: element.scrollHeight,
+        clientWidth: element.clientWidth,
+        clientHeight: element.clientHeight,
+        scrollLeft: element.scrollLeft,
+        scrollTop: element.scrollTop,
```

Comment:
Do we actually need to copy each prop? What about only using the element?

Reply:
Copy/paste from react-swipeable-views. I have no idea. :sweat_smile: Maybe because the element might change in the meantime (it won't, because the meantime is some nanoseconds without the event loop being involved)?

Reply:
@leMaik No idea either, I suspect it was simpler to debug/test. I think that we should remove it.

Reply:
Still works when replacing this with `domTreeShapes.push(element)`. :man_shrugging: